### PR TITLE
Detect image-generation chats from ui-component message parts

### PR DESCRIPTION
**Pull request by GerardPaligot**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1212015278241917/task/1214918311210465?focus=true 

### Description

ImageGeneration chat detection moves from `DuckAiChat.model` to a structural rule on the chat message tree (assistant message whose `parts[]` carries a `ui-component` / `generate-image` entry). The resulting per-type leading icon (Discussion / ImageGeneration / Voice) now drives both the chat history screen and the omnibar chat-history suggestions; on suggestions it sits behind the new `chatSuggestionTypeIcon` toggle (default TRUE) so the propagation can be rolled back independently.

### Steps to test this PR

> [!NOTE]
> Prerequisites:
> - [ ] Install Internal Debug.
> - [ ] `duckChat`: confirm `useNativeStorageChatData` is ON and `chatSuggestionTypeIcon` is ON.
> - [ ] In Duck.ai, create three chats: (a) a normal text discussion, (b) an image-generation chat (ask Duck.ai to draw something so the FE produces the `ui-component` / `generate-image` part), (c) a voice chat.

_Happy path — icons by chat type on both surfaces_

- [ ] Open the chat history screen from the browser menu → confirm each row's leading icon matches its chat type (chat bubble for text, picture for image-gen, mic for voice).
- [ ] Focus the omnibar so chat-history suggestions appear → confirm each suggestion row's leading icon matches the same chat type as the corresponding row on the chat history screen.

_Guards — pinned variants_

- [ ] Pin the image-generation chat from Duck.ai → re-open both surfaces → confirm both show the pinned ImageGeneration icon variant.

_Flag rollback (suggestions only)_

- [ ] In Developer Settings, flip `duckChat.chatSuggestionTypeIcon` to OFF.
- [ ] Trigger chat-history suggestions in the omnibar → confirm every row renders the Discussion icon (chat bubble), regardless of chat type.
- [ ] Open the chat history screen → confirm it still renders per-type icons (the flag does not affect the history screen).

### UI changes
| Before  | After |
| ------ | ----- |
image | image

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes how chat type is derived by parsing FE-owned chat JSON (including iterating message parts) and uses that classification to drive UI rendering behind a new toggle for suggestions.
> 
> **Overview**
> **Chat type classification is refactored and surfaced in UI.** `DuckAiChat` now includes derived flags (`isImageGeneration`, `isVoice`), with image-generation detected by scanning assistant `messages[].parts[]` for a `ui-component` named `generate-image`, and `ChatType` mapping moved into `impl.models`.
> 
> **Per-type leading icons are now used consistently.** Chat history rows and omnibar chat-history suggestions render icons via a shared `ChatType.iconRes(pinned)` helper; suggestions additionally carry a `type` field and are gated by a new `duckChat.chatSuggestionTypeIcon` toggle (default `true`) that can force the legacy Discussion icon.
> 
> Tests are updated/added to validate the new classification rules and toggle behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bdde78cbfb4bf779681c5d6126f72e90bfe6970a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by GerardPaligot** [comment]
* **#8623** <a href="https://app.graphite.com/github/pr/duckduckgo/Android/8623?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/duckduckgo/Android/8623?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/feature/DuckChatFeature.kt [hunk 1]
```diff
@@ -234,6 +234,14 @@ interface DuckChatFeature {
     @Toggle.DefaultValue(DefaultFeatureValue.INTERNAL)
     fun deleteFromAutocomplete(): Toggle
 
+    /**
+     * @return `true` when chat-suggestion rows in the omnibar autocomplete render a per-type leading icon
+     * (Discussion / ImageGeneration / Voice) instead of the default Discussion icon.
+     * If the remote feature is not present defaults to `true`.
+     */
+    @Toggle.DefaultValue(DefaultFeatureValue.TRUE)
+    fun chatSuggestionTypeIcon(): Toggle
```
Comment (malmstein) [inline duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/feature/DuckChatFeature.kt:243]: this toggle gates the per-type icon in `ChatSuggestionsNativeReader`, but `ChatHistoryRepository.toChatHistoryItem` calls `chat.toChatType()` unconditionally — if the new message-tree derivation regresses we'd only be able to kill it for the suggestion rows in the omnibar. is the asymmetry intentional? the toggle name is suggestion-specific so maybe yes, but if the kill switch exists as defence-in-depth against the new derivation, it feels like both surfaces should be covered.

**Review by malmstein — APPROVED** [review]
smoke-tested the APK on the Pixel 9 and the per-type icons render correctly in both chat history and the omnibar suggestions, including the mixed text/image-gen chats. nice refactor moving `ChatType` to `models/` and unifying `iconRes` across both viewholders — and the store tests cover the JSON walk edge cases thoroughly (assistant-only, ui-component type guard, missing messages/parts). one concern on the kill switch's asymmetric coverage between chat suggestions and chat history.

File: duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/feature/DuckChatFeature.kt (hunk 1 above)
Comment (GerardPaligot) [reply duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/feature/DuckChatFeature.kt:243]: Yes, it is intentional. This decision has been taken in the [Asana task](https://app.asana.com/1/137249556945/project/488551667048375/task/1214918311210465?focus=true)
